src/main/java/org/motechproject/newebodac/repository/custom/impl/FieldConfigRepositoryImpl.java

```diff
@@ -0,0 +1,41 @@
+package org.motechproject.newebodac.repository.custom.impl;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.motechproject.newebodac.domain.BaseEntity;
+import org.motechproject.newebodac.domain.enums.EntityType;
+import org.motechproject.newebodac.exception.NewEbodacException;
+import org.motechproject.newebodac.repository.custom.FieldConfigRepositoryCustom;
+
+public class FieldConfigRepositoryImpl implements FieldConfigRepositoryCustom {
+
+  private final EntityManager em;
+  private final CriteriaBuilder cb;
+  
+  public FieldConfigRepositoryImpl(EntityManager em) {
+    this.em = em;
+    this.cb = em.getCriteriaBuilder();
+  }
+
+  /** Checks if value should be unique, then if already exists. */
+
+  @Override
+
+
+  public void checkIfUnique(EntityType entityType, BaseEntity entity,
+      String fieldName, Object fieldValue) {
+    CriteriaQuery<Long> query = cb.createQuery(Long.class);
+    Root root = query.from(entityType.getEntityClass());
+    query.select(root).where(cb.equal(root.get(fieldName), fieldValue));
```

Comment:
It was not enough, because during new entity creation the value of id was null and the new entity with the same value was created. Adding if statement solved the problem.